src/Modifiers/CoreModifiers.php

```diff
@@ -1463,6 +1463,25 @@ public function macro($value, $params, $context)
         }, $value);
     }
 
+    /**
+     * Wraps matched words with <mark> tags.
+     *
+     * @param $value
+     * @param  array  $params
+     * @return string
+     */
+    public function mark($value, $params, $context)
+    {
+        $words = $params[0] ?? $context['get']['q'] ?? null;
```

Comment:
If you don't pass an argument, and there's no `q` query string, it wraps every character in a mark. You should return early with the original value here in that case.